torch/_inductor/scheduler.py

```diff
@@ -66,17 +67,47 @@
 from .virtualized import V
 
 
+if TYPE_CHECKING:
+    from .memory import InputBuffer
+
 log = logging.getLogger(__name__)
 fusion_log = torch._logging.getArtifactLogger(__name__, "fusion")
 loop_ordering_log = torch._logging.getArtifactLogger(__name__, "loop_ordering")
 
 
+@dataclasses.dataclass
+class MemoryPlanningInfoForBuffer:
+    size_alloc: int = 0
+    size_free: int = 0
+    outdegree: int = 0
+    succ_nodes: OrderedSet[BaseSchedulerNode] = dataclasses.field(
+        default_factory=OrderedSet
+    )
+
+
+@dataclasses.dataclass
+class MemoryPlanningInfoForNode:
+    pred_buffers: List[Union[SchedulerBuffer, InputBuffer]] = dataclasses.field(
+        default_factory=list
+    )
+    pred_nodes: List[BaseSchedulerNode] = dataclasses.field(default_factory=list)
+    succ_nodes: List[BaseSchedulerNode] = dataclasses.field(default_factory=list)
+    indegree: int = 0
+    index: int = 0
+    memory_to_free: int = 0
+    size: int = 0
+    measure: int = 0
```

Comment:
maybe we can import these two dataclasses from memory.py as well?

Reply:
I can't believe how much time I spent on this, fixing linter and circular import issues 😅 but finally made it work. I moved these two classes to the `memory.py` file now.